apps/web-portal-client/lib/components/HomePage.jsx

```diff
@@ -1,17 +1,16 @@
-import React, {useEffect} from 'react'
+import React, {useContext} from 'react'
+import {useFetchData} from '../hooks'
+import {AppContext} from '../context'
 
 import Spinner from './Spinner'
 
 import {UserProfile} from './UserProfile'
 import {PreviewCard} from './PreviewCard'
 import {Content, Sidebar, Layout} from './layout'
 
-import {usePreviews} from '../features/home/hooks'
-
 export function HomePage() {
-  const [{previews}, {isLoading, isError, doFetch}] = usePreviews()
-
-  useEffect(() => doFetch(), [])
+  const {api} = useContext(AppContext)
+  const {data: previews, isLoading, isError} = useFetchData(api.preview.findAll())
```

Comment:
It's not an efficient usage of hook api, code above is doing 3 requests instead of one

<img width="1467" alt="Screenshot 2021-03-09 at 20 05 59" src="https://user-images.githubusercontent.com/75000/110517366-714a7980-8113-11eb-9ce8-243e4b139cf1.png">